Geometry_Engine/Query/IsContaining.cs

```diff
@@ -709,6 +709,158 @@ public static bool IsContaining(this Cuboid cuboid, IGeometry geometry, bool acc
+        /**** Private Methods - Closed Volume Polylines ****/
+        /***************************************************/
+
+        [Description("Checks if a point lies within a collection of polylines describing a closed volume using their primitive planes and bounds.")]
+        [Input("closedVolume", "A collection of Polylines outlining each of the faces of the closed volume to check with.")]
+        [Input("planes", "Planes corresponding to each panel for intersection calculations.")]
+        [Input("boundingBox", "The bounding box of the collection of panels.")]
+        [Input("point", "The point to check to see if it is contained within the bounds of the panels.")]
+        [Input("acceptOnEdges", "Decide whether to allow the point to sit on the edge of the panel, default false.")]
+        [Input("tolerance", "Distance tolerance to use to determine intersections.")]
+        [Output("isContaining", "True if the point is contained within the bounds of the panels, false if it is not for each point provided.")]
+        private static bool IsContaining(this List<Polyline> closedVolume, List<Plane> planes, BoundingBox boundingBox, Point point, bool acceptOnEdges = false, double tolerance = BH.oM.Geometry.Tolerance.Distance)
+        {
+            //Return if point is null even without checking boundingBox.IsContaining(point)
+            if (point == null)
+            {
+                BH.Engine.Base.Compute.RecordError("Cannot query if a collection of panels contains a point if the point is null.");
+                return false;
+            }
+
+            if (planes.Any(x => x == null))
+            {
+                BH.Engine.Base.Compute.RecordError("Cannot query if a provided plane is null.");
+                return false;
+            }
+
+            if (!BH.Engine.Geometry.Query.IsContaining(boundingBox, point, true, tolerance))
+                return false;
+
+            Vector rVec;
+            bool isInPlane = false;
+            do
+            {
+                int seed = 1;
+                 rVec = BH.Engine.Geometry.Create.RandomVector(seed); 
+                
+                isInPlane = false;
+                foreach (Plane p in planes)
+                    isInPlane = isInPlane || rVec.IsInPlane(p, tolerance);
+
+                if (isInPlane)
+                {
+                    seed += 1;
+                }
+            }
+            while (isInPlane);
+
+            //Use a length longer than the longest side in the bounding box. Change to infinite line?
+            Line line = new Line() { Start = point, End = point.Translate(rVec * (((boundingBox.Max - boundingBox.Min).Length()) * 10)) };
+
+            //Check intersections
+            List<Point> intersectPoints = new List<Point>();
+
+            for (int x = 0; x < planes.Count; x++)
+            {
+                Point intPt = PlaneIntersection(line, planes[x], true, tolerance);
```

Comment:
Are you aware your line will not be treated as infinite here? To achieve that, you would first need to set `Infinite = true`. I think we do not need it here, but just FYI

Reply:
Nope, at least from how I read the code of `PlaneIntersection` query: you need to have both `true` parameter and `line.Infinite` set to `true`

Reply:
Resolved per latest commit from @pawelbaran